apps/sms/services/js/moz_settings/moz_settings_shim.js

```diff
@@ -0,0 +1,65 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* global BridgeServiceMixin */
+
+/* exported MozSettingsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozSettings API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-settings-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['get', 'set']);
+
+var mozSettings = null;
+
+var MozSettingsShim = {
+  init(settings) {
+    if (!settings) {
+      return;
+    }
+    mozSettings = settings;
+
+    this.initService();
+  },
+
+  /* Methods */
+
+  /**
+   * Shim for mozSettings.createLock().get API.
+   * @param {String} key Key name for Settings.
+   * @returns {Promise} Promise that return the settings value from settings DB.
+   */
+  get(key) {
+    return mozSettings.createLock().get(key);
+  },
+
+  /**
+   * Shim for mozSettings.createLock().set API.
+   * @param {Object} settings Settings key and value for settings. It can have
+   *  several key/value pairs inside.
+   * @returns {Promise} Promise that return saved result.
+   */
+  set(settings) {
+    return mozSettings.createLock().set(settings);
+  }
+};
+
+exports.MozSettingsShim = Object.seal(
+  BridgeServiceMixin.mixin(
```

Comment:
Maybe iframe contains shims forgot to load the bridge or stream, and it will cause shims to load the scripts by importScripts API that not existed in window context. But you still still see the error and figure out the problem anyway, so it's really minor.